plugins/inputs/linux_cpu/linux_cpu.go

```diff
@@ -0,0 +1,188 @@
+// +build linux
+
+package linux_cpu
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const defaultHostSys = "/sys"
+
+type LinuxCPU struct {
+	Log             telegraf.Logger `toml:"-"`
+	PathSysfs       string          `toml:"host_sys"`
+	GatherCPUFreq   bool            `toml:"gather_cpufreq"`
+	GatherThrottles bool            `toml:"gather_throttles"`
```

Comment:
Please, avoid boolean fields when a string or enumerated type would be better for future extension. Lots of boolean fields also make the code difficult to maintain.

(taken from: https://github.com/influxdata/telegraf/wiki/Review#reviewing-plugin-code )